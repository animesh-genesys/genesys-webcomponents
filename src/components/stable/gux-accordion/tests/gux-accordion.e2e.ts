import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

describe('gux-accordion', () => {
  it('renders', async () => {
    const html = `
      <gux-accordion>
        <gux-accordion-section>
          <h2 slot="header">First Section</h2>
          <div slot="content">Sample Content</div>
        </gux-accordion-section>
        <gux-accordion-section arrow-position="beside-text">
          <h2 slot="header">Second Section</h2>
          <div slot="content">Sample Content</div>
        </gux-accordion-section>
        <gux-accordion-section arrow-position="before-text">
          <h2 slot="header">Third Section</h2>
          <div slot="content">Sample Content</div>
        </gux-accordion-section>
        <gux-accordion-section content-layout="custom">
          <h2 slot="header">Fourth Section</h2>
          <div slot="content">Sample Content</div>
        </gux-accordion-section>
        <gux-accordion-section content-layout="text">
          <h2 slot="header">Fifth Section</h2>
          <div slot="content">Sample Content</div>
        </gux-accordion-section>
        <gux-accordion-section disabled>
          <h2 slot="header">Sixth Section</h2>
          <h3 slot="subheader">This is the subheader</h3>
          <div slot="content">Sample Content</div>
        </gux-accordion-section>
        <gux-accordion-section>
          <h2 slot="header">Seventh Section</h2>
          <h3 slot="subheader">This is the subheader</h3>
          <gux-icon
            slot="icon"
            icon-name="add-user"
            screenreader-text="add John Smith to contact list"
          ></gux-icon>
          <div slot="content">Sample Content</div>
        </gux-accordion-section>
        <gux-accordion-section reverse-headings="true">
          <h2 slot="header">Eighth Section</h2>
          <h3 slot="subheader">This is the subheader</h3>
          <gux-icon
            slot="icon"
            icon-name="add-user"
            screenreader-text="add John Smith to contact list"
          ></gux-icon>
          <div slot="content">Sample Content</div>
        </gux-accordion-section>
      </gux-accordion>
      `;
    const page = await newSparkE2EPage({ html });
    const element = await page.find('gux-accordion');

    await a11yCheck(page);

    expect(element.outerHTML).toMatchSnapshot();
  });
});
