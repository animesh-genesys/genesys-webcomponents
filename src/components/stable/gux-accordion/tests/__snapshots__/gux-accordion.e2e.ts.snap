// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-accordion renders 1`] = `"<gux-accordion lang=\\"en\\" text=\\"Primary\\" accent=\\"primary\\" hydrated=\\"\\"> <gux-accordion hydrated=\\"\\"> <gux-accordion-section hydrated=\\"\\"> <h2 slot=\\"header\\">First Section</h2> <div slot=\\"content\\">Sample Content</div> </gux-accordion-section> <gux-accordion-section hydrated=\\"\\"> <h2 slot=\\"header\\">Second Section</h2> <div slot=\\"content\\">Sample Content</div> </gux-accordion-section> <gux-accordion-section hydrated=\\"\\"> <h2 slot=\\"header\\">Third Section</h2> <div slot=\\"content\\">Sample Content</div> </gux-accordion-section> </gux-accordion> </gux-accordion>"`;
